src/web/modules/payouts/reconcile.ts

```diff
@@ -0,0 +1,167 @@
+
+  if (match.length !== 1) {
+    throw new Error(`Unable to process refunds for payment ${payId} (unknown number of matches)`)
+  }
+  const payRefund = match.pop()
+
+  return {
+    payId: payRefund.refund_id,
+    payReference: charge.reference,
+    type: PaymentType.REFUND,
+    transactionDate: new Date(payRefund.created_date),
+    amount: payRefund.amount,
+    fee: 0,
+    net: payRefund.amount,
+    refundForPayId: refunds.payment_id,
+    // @ts-ignore
+    gatewayId: refund.source.id
+  }
+}
+
+const verifyReconciledTotals = async function verifyReconciledTotals(
+  payout: Stripe.payouts.IPayout,
+  transactions: PayTransactionCSVEntity[]
+): Promise<void> {
+  const grouped = _.groupBy(transactions, 'type')
+  const reconciledAmount = _.subtract(
+    _.sumBy(grouped[PaymentType.PAYMENT], 'net'),
+    _.sumBy(grouped[PaymentType.REFUND], 'net')
+  )
+
+  if (reconciledAmount !== payout.amount) {
+    throw new Error('Unable to exactly match Stripe payout to GOV.UK Pay transactions (amount)')
+  }
+}
+
+const reconcileStripePayTransactions = async function reconcileStripePayTransactions(
+  gatewayAccountId: string,
+  payout: Stripe.payouts.IPayout,
+  transactions: Stripe.IList<Stripe.balance.IBalanceTransaction>
+): Promise<PayTransactionCSVEntity[]> {
+  const grouped = _.groupBy(transactions.data, 'type')
+  const { transfer: refunds, payment: payments } = grouped
+  const reconciled: PayTransactionCSVEntity[] = []
+
+  const unknownKeys = Object.keys(grouped).filter(key => ![ 'payment', 'transfer', 'payout' ].includes(key))
+  if (unknownKeys.length) {
+    throw new Error('Unable to exactly match Stripe payout to GOV.UK Pay transactions (unknown types)')
+  }
+
+  if (refunds) {
+    for (const refund of refunds) {
+      reconciled.push(await reconcileRefund(gatewayAccountId, refund))
+    }
+  }
+
+  if (payments) {
+    for (const payment of payments) {
+      reconciled.push(await reconcilePayment(gatewayAccountId, payment))
+    }
+  }
+
+  await verifyReconciledTotals(payout, reconciled)
+  return reconciled
+}
+
+export async function buildPayoutCSVReport(
+  gatewayAccountId: string,
+  payout: Stripe.payouts.IPayout,
+  stripeAccountId: string
+): Promise<string> {
+  // @TODO(sfount) pages.js equivalent to not be limited by 100 transactions
+  const transactions = await getTransactionsForPayout(stripeAccountId, payout)
+  const payTransactions = await reconcileStripePayTransactions(
+    gatewayAccountId,
+    payout,
+    transactions
+  )
+  return renderCSV(payTransactions, payout)
+}
+
+export async function getPayoutsForAccount(
+  stripeAccountId: string,
+  startingAfter: string,
+  endingBefore: string
+): Promise<Stripe.IList<Stripe.payouts.IPayout>> {
+  const limitPayoutPageSize = 10
+  const options: Stripe.payouts.IPayoutListOptions = {
+    limit: limitPayoutPageSize,
+    ...startingAfter && { starting_after: startingAfter },
+    ...endingBefore && { ending_before: endingBefore }
```

Comment:
Why spread operator? I don't know this use case. Could you explain why is it required?

Reply:
Replaces:
```js
if (startingAfter) { 
  options.starting_after = startingAfter
}
```

Not 100% required but succinct!

Reply:
Brackets would help with readability.
```js
...(endingBefore && { ending_before: endingBefore })
```